Traceback (most recent call last):
  File "/Users/aet/mambaforge/envs/codeforecon/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/aet/mambaforge/envs/codeforecon/lib/python3.10/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/aet/mambaforge/envs/codeforecon/lib/python3.10/site-packages/jupyter_core/utils/__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
  File "/Users/aet/mambaforge/envs/codeforecon/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/Users/aet/mambaforge/envs/codeforecon/lib/python3.10/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/aet/mambaforge/envs/codeforecon/lib/python3.10/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/aet/mambaforge/envs/codeforecon/lib/python3.10/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import matplotlib_inline.backend_inline
import matplotlib.pyplot as plt

# Plot settings
plt.style.use(
    "https://github.com/aeturrell/coding-for-economists/raw/main/plot_style.txt"
)
matplotlib_inline.backend_inline.set_matplotlib_formats("svg")
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mTimeoutError[0m                              Traceback (most recent call last)
File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/urllib/request.py:1348[0m, in [0;36mAbstractHTTPHandler.do_open[0;34m(self, http_class, req, **http_conn_args)[0m
[1;32m   1347[0m [38;5;28;01mtry[39;00m:
[0;32m-> 1348[0m     [43mh[49m[38;5;241;43m.[39;49m[43mrequest[49m[43m([49m[43mreq[49m[38;5;241;43m.[39;49m[43mget_method[49m[43m([49m[43m)[49m[43m,[49m[43m [49m[43mreq[49m[38;5;241;43m.[39;49m[43mselector[49m[43m,[49m[43m [49m[43mreq[49m[38;5;241;43m.[39;49m[43mdata[49m[43m,[49m[43m [49m[43mheaders[49m[43m,[49m
[1;32m   1349[0m [43m              [49m[43mencode_chunked[49m[38;5;241;43m=[39;49m[43mreq[49m[38;5;241;43m.[39;49m[43mhas_header[49m[43m([49m[38;5;124;43m'[39;49m[38;5;124;43mTransfer-encoding[39;49m[38;5;124;43m'[39;49m[43m)[49m[43m)[49m
[1;32m   1350[0m [38;5;28;01mexcept[39;00m [38;5;167;01mOSError[39;00m [38;5;28;01mas[39;00m err: [38;5;66;03m# timeout error[39;00m

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/http/client.py:1283[0m, in [0;36mHTTPConnection.request[0;34m(self, method, url, body, headers, encode_chunked)[0m
[1;32m   1282[0m [38;5;250m[39m[38;5;124;03m"""Send a complete request to the server."""[39;00m
[0;32m-> 1283[0m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_send_request[49m[43m([49m[43mmethod[49m[43m,[49m[43m [49m[43murl[49m[43m,[49m[43m [49m[43mbody[49m[43m,[49m[43m [49m[43mheaders[49m[43m,[49m[43m [49m[43mencode_chunked[49m[43m)[49m

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/http/client.py:1329[0m, in [0;36mHTTPConnection._send_request[0;34m(self, method, url, body, headers, encode_chunked)[0m
[1;32m   1328[0m     body [38;5;241m=[39m _encode(body, [38;5;124m'[39m[38;5;124mbody[39m[38;5;124m'[39m)
[0;32m-> 1329[0m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mendheaders[49m[43m([49m[43mbody[49m[43m,[49m[43m [49m[43mencode_chunked[49m[38;5;241;43m=[39;49m[43mencode_chunked[49m[43m)[49m

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/http/client.py:1278[0m, in [0;36mHTTPConnection.endheaders[0;34m(self, message_body, encode_chunked)[0m
[1;32m   1277[0m     [38;5;28;01mraise[39;00m CannotSendHeader()
[0;32m-> 1278[0m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_send_output[49m[43m([49m[43mmessage_body[49m[43m,[49m[43m [49m[43mencode_chunked[49m[38;5;241;43m=[39;49m[43mencode_chunked[49m[43m)[49m

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/http/client.py:1038[0m, in [0;36mHTTPConnection._send_output[0;34m(self, message_body, encode_chunked)[0m
[1;32m   1037[0m [38;5;28;01mdel[39;00m [38;5;28mself[39m[38;5;241m.[39m_buffer[:]
[0;32m-> 1038[0m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43msend[49m[43m([49m[43mmsg[49m[43m)[49m
[1;32m   1040[0m [38;5;28;01mif[39;00m message_body [38;5;129;01mis[39;00m [38;5;129;01mnot[39;00m [38;5;28;01mNone[39;00m:
[1;32m   1041[0m 
[1;32m   1042[0m     [38;5;66;03m# create a consistent interface to message_body[39;00m

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/http/client.py:976[0m, in [0;36mHTTPConnection.send[0;34m(self, data)[0m
[1;32m    975[0m [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39mauto_open:
[0;32m--> 976[0m     [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mconnect[49m[43m([49m[43m)[49m
[1;32m    977[0m [38;5;28;01melse[39;00m:

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/http/client.py:1448[0m, in [0;36mHTTPSConnection.connect[0;34m(self)[0m
[1;32m   1446[0m [38;5;124m"[39m[38;5;124mConnect to a host on a given (SSL) port.[39m[38;5;124m"[39m
[0;32m-> 1448[0m [38;5;28;43msuper[39;49m[43m([49m[43m)[49m[38;5;241;43m.[39;49m[43mconnect[49m[43m([49m[43m)[49m
[1;32m   1450[0m [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39m_tunnel_host:

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/http/client.py:942[0m, in [0;36mHTTPConnection.connect[0;34m(self)[0m
[1;32m    941[0m sys[38;5;241m.[39maudit([38;5;124m"[39m[38;5;124mhttp.client.connect[39m[38;5;124m"[39m, [38;5;28mself[39m, [38;5;28mself[39m[38;5;241m.[39mhost, [38;5;28mself[39m[38;5;241m.[39mport)
[0;32m--> 942[0m [38;5;28mself[39m[38;5;241m.[39msock [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_create_connection[49m[43m([49m
[1;32m    943[0m [43m    [49m[43m([49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mhost[49m[43m,[49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mport[49m[43m)[49m[43m,[49m[43m [49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mtimeout[49m[43m,[49m[43m [49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43msource_address[49m[43m)[49m
[1;32m    944[0m [38;5;66;03m# Might fail in OSs that don't implement TCP_NODELAY[39;00m

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/socket.py:845[0m, in [0;36mcreate_connection[0;34m(address, timeout, source_address)[0m
[1;32m    844[0m [38;5;28;01mtry[39;00m:
[0;32m--> 845[0m     [38;5;28;01mraise[39;00m err
[1;32m    846[0m [38;5;28;01mfinally[39;00m:
[1;32m    847[0m     [38;5;66;03m# Break explicitly a reference cycle[39;00m

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/socket.py:833[0m, in [0;36mcreate_connection[0;34m(address, timeout, source_address)[0m
[1;32m    832[0m     sock[38;5;241m.[39mbind(source_address)
[0;32m--> 833[0m [43msock[49m[38;5;241;43m.[39;49m[43mconnect[49m[43m([49m[43msa[49m[43m)[49m
[1;32m    834[0m [38;5;66;03m# Break explicitly a reference cycle[39;00m

[0;31mTimeoutError[0m: [Errno 60] Operation timed out

During handling of the above exception, another exception occurred:

[0;31mURLError[0m                                  Traceback (most recent call last)
File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/site-packages/matplotlib/style/core.py:166[0m, in [0;36muse[0;34m(style)[0m
[1;32m    165[0m [38;5;28;01mtry[39;00m:
[0;32m--> 166[0m     style [38;5;241m=[39m [43m_rc_params_in_file[49m[43m([49m[43mstyle[49m[43m)[49m
[1;32m    167[0m [38;5;28;01mexcept[39;00m [38;5;167;01mIOError[39;00m [38;5;28;01mas[39;00m err:

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/site-packages/matplotlib/__init__.py:854[0m, in [0;36m_rc_params_in_file[0;34m(fname, transform, fail_on_error)[0m
[1;32m    853[0m rc_temp [38;5;241m=[39m {}
[0;32m--> 854[0m [38;5;28;01mwith[39;00m _open_file_or_url(fname) [38;5;28;01mas[39;00m fd:
[1;32m    855[0m     [38;5;28;01mtry[39;00m:

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/contextlib.py:135[0m, in [0;36m_GeneratorContextManager.__enter__[0;34m(self)[0m
[1;32m    134[0m [38;5;28;01mtry[39;00m:
[0;32m--> 135[0m     [38;5;28;01mreturn[39;00m [38;5;28;43mnext[39;49m[43m([49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mgen[49m[43m)[49m
[1;32m    136[0m [38;5;28;01mexcept[39;00m [38;5;167;01mStopIteration[39;00m:

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/site-packages/matplotlib/__init__.py:827[0m, in [0;36m_open_file_or_url[0;34m(fname)[0m
[1;32m    824[0m     _log[38;5;241m.[39mdebug(
[1;32m    825[0m         [38;5;124m"[39m[38;5;124mCould not get certifi ssl context, https may not work.[39m[38;5;124m"[39m
[1;32m    826[0m     )
[0;32m--> 827[0m [38;5;28;01mwith[39;00m [43murllib[49m[38;5;241;43m.[39;49m[43mrequest[49m[38;5;241;43m.[39;49m[43murlopen[49m[43m([49m[43mfname[49m[43m,[49m[43m [49m[43mcontext[49m[38;5;241;43m=[39;49m[43mssl_ctx[49m[43m)[49m [38;5;28;01mas[39;00m f:
[1;32m    828[0m     [38;5;28;01myield[39;00m (line[38;5;241m.[39mdecode([38;5;124m'[39m[38;5;124mutf-8[39m[38;5;124m'[39m) [38;5;28;01mfor[39;00m line [38;5;129;01min[39;00m f)

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/urllib/request.py:216[0m, in [0;36murlopen[0;34m(url, data, timeout, cafile, capath, cadefault, context)[0m
[1;32m    215[0m     opener [38;5;241m=[39m _opener
[0;32m--> 216[0m [38;5;28;01mreturn[39;00m [43mopener[49m[38;5;241;43m.[39;49m[43mopen[49m[43m([49m[43murl[49m[43m,[49m[43m [49m[43mdata[49m[43m,[49m[43m [49m[43mtimeout[49m[43m)[49m

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/urllib/request.py:519[0m, in [0;36mOpenerDirector.open[0;34m(self, fullurl, data, timeout)[0m
[1;32m    518[0m sys[38;5;241m.[39maudit([38;5;124m'[39m[38;5;124murllib.Request[39m[38;5;124m'[39m, req[38;5;241m.[39mfull_url, req[38;5;241m.[39mdata, req[38;5;241m.[39mheaders, req[38;5;241m.[39mget_method())
[0;32m--> 519[0m response [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_open[49m[43m([49m[43mreq[49m[43m,[49m[43m [49m[43mdata[49m[43m)[49m
[1;32m    521[0m [38;5;66;03m# post-process response[39;00m

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/urllib/request.py:536[0m, in [0;36mOpenerDirector._open[0;34m(self, req, data)[0m
[1;32m    535[0m protocol [38;5;241m=[39m req[38;5;241m.[39mtype
[0;32m--> 536[0m result [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_call_chain[49m[43m([49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mhandle_open[49m[43m,[49m[43m [49m[43mprotocol[49m[43m,[49m[43m [49m[43mprotocol[49m[43m [49m[38;5;241;43m+[39;49m
[1;32m    537[0m [43m                          [49m[38;5;124;43m'[39;49m[38;5;124;43m_open[39;49m[38;5;124;43m'[39;49m[43m,[49m[43m [49m[43mreq[49m[43m)[49m
[1;32m    538[0m [38;5;28;01mif[39;00m result:

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/urllib/request.py:496[0m, in [0;36mOpenerDirector._call_chain[0;34m(self, chain, kind, meth_name, *args)[0m
[1;32m    495[0m func [38;5;241m=[39m [38;5;28mgetattr[39m(handler, meth_name)
[0;32m--> 496[0m result [38;5;241m=[39m [43mfunc[49m[43m([49m[38;5;241;43m*[39;49m[43margs[49m[43m)[49m
[1;32m    497[0m [38;5;28;01mif[39;00m result [38;5;129;01mis[39;00m [38;5;129;01mnot[39;00m [38;5;28;01mNone[39;00m:

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/urllib/request.py:1391[0m, in [0;36mHTTPSHandler.https_open[0;34m(self, req)[0m
[1;32m   1390[0m [38;5;28;01mdef[39;00m [38;5;21mhttps_open[39m([38;5;28mself[39m, req):
[0;32m-> 1391[0m     [38;5;28;01mreturn[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mdo_open[49m[43m([49m[43mhttp[49m[38;5;241;43m.[39;49m[43mclient[49m[38;5;241;43m.[39;49m[43mHTTPSConnection[49m[43m,[49m[43m [49m[43mreq[49m[43m,[49m
[1;32m   1392[0m [43m        [49m[43mcontext[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_context[49m[43m,[49m[43m [49m[43mcheck_hostname[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_check_hostname[49m[43m)[49m

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/urllib/request.py:1351[0m, in [0;36mAbstractHTTPHandler.do_open[0;34m(self, http_class, req, **http_conn_args)[0m
[1;32m   1350[0m [38;5;28;01mexcept[39;00m [38;5;167;01mOSError[39;00m [38;5;28;01mas[39;00m err: [38;5;66;03m# timeout error[39;00m
[0;32m-> 1351[0m     [38;5;28;01mraise[39;00m URLError(err)
[1;32m   1352[0m r [38;5;241m=[39m h[38;5;241m.[39mgetresponse()

[0;31mURLError[0m: <urlopen error [Errno 60] Operation timed out>

The above exception was the direct cause of the following exception:

[0;31mOSError[0m                                   Traceback (most recent call last)
Cell [0;32mIn[1], line 5[0m
[1;32m      2[0m [38;5;28;01mimport[39;00m [38;5;21;01mmatplotlib[39;00m[38;5;21;01m.[39;00m[38;5;21;01mpyplot[39;00m [38;5;28;01mas[39;00m [38;5;21;01mplt[39;00m
[1;32m      4[0m [38;5;66;03m# Plot settings[39;00m
[0;32m----> 5[0m [43mplt[49m[38;5;241;43m.[39;49m[43mstyle[49m[38;5;241;43m.[39;49m[43muse[49m[43m([49m
[1;32m      6[0m [43m    [49m[38;5;124;43m"[39;49m[38;5;124;43mhttps://github.com/aeturrell/coding-for-economists/raw/main/plot_style.txt[39;49m[38;5;124;43m"[39;49m
[1;32m      7[0m [43m)[49m
[1;32m      8[0m matplotlib_inline[38;5;241m.[39mbackend_inline[38;5;241m.[39mset_matplotlib_formats([38;5;124m"[39m[38;5;124msvg[39m[38;5;124m"[39m)

File [0;32m~/mambaforge/envs/codeforecon/lib/python3.10/site-packages/matplotlib/style/core.py:168[0m, in [0;36muse[0;34m(style)[0m
[1;32m    166[0m         style [38;5;241m=[39m _rc_params_in_file(style)
[1;32m    167[0m     [38;5;28;01mexcept[39;00m [38;5;167;01mIOError[39;00m [38;5;28;01mas[39;00m err:
[0;32m--> 168[0m         [38;5;28;01mraise[39;00m [38;5;167;01mIOError[39;00m(
[1;32m    169[0m             [38;5;124mf[39m[38;5;124m"[39m[38;5;132;01m{[39;00mstyle[38;5;132;01m!r}[39;00m[38;5;124m is not a valid package style, path of style [39m[38;5;124m"[39m
[1;32m    170[0m             [38;5;124mf[39m[38;5;124m"[39m[38;5;124mfile, URL of style file, or library style name (library [39m[38;5;124m"[39m
[1;32m    171[0m             [38;5;124mf[39m[38;5;124m"[39m[38;5;124mstyles are listed in `style.available`)[39m[38;5;124m"[39m) [38;5;28;01mfrom[39;00m [38;5;21;01merr[39;00m
[1;32m    172[0m filtered [38;5;241m=[39m {}
[1;32m    173[0m [38;5;28;01mfor[39;00m k [38;5;129;01min[39;00m style:  [38;5;66;03m# don't trigger RcParams.__getitem__('backend')[39;00m

[0;31mOSError[0m: 'https://github.com/aeturrell/coding-for-economists/raw/main/plot_style.txt' is not a valid package style, path of style file, URL of style file, or library style name (library styles are listed in `style.available`)

